taichi/program/kernel.cpp

```diff
@@ -65,6 +66,29 @@ Kernel::Kernel(Program &program,
     compile();
 }
 
+Kernel::Kernel(Program &program,
+               std::unique_ptr<IRNode> &&ir,
+               const std::string &primal_name,
+               bool grad)
+    : ir(std::move(ir)), program(program), lowered(false), grad(grad) {
+  is_accessor = false;
+  is_evaluator = false;
+  compiled = nullptr;
+  ir_is_ast = false;  // CHI IR
+  this->ir->as<Block>()->kernel = this;
+
+  arch = program.config.arch;
+
+  if (!grad) {
+    name = primal_name;
+  } else {
+    name = primal_name + "_grad";
+  }
+
+  if (!program.config.lazy_compilation)
+    compile();
+}
```

Comment:
This part seems to slightly duplicate the other constructor - but assuming this ctor is just temporary, this should be fine!